File: 897. Increasing Order Search Tree/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func inorder(node *TreeNode, nodes []int) []int {
    if node != nil {
        nodes = inorder(node.Left, nodes)
        nodes = append(nodes, node.Val)
        nodes = inorder(node.Right, nodes)
    }
    return nodes
}

func increasingBST(root *TreeNode) *TreeNode {
    nodes := []int{}
    nodes = inorder(root,nodes)
    cur := &TreeNode{nodes[0], nil, nil}
    ans := cur
    for _,n := range nodes[1:] {
        cur.Right = &TreeNode{n, nil, nil}
        cur = cur.Right
    }
    return ans
    
}

File: 897. Increasing Order Search Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    """将BST转为全右子树结构，先序遍历获得值列表，再重建"""

    def increasingBST(self, root: TreeNode) -> TreeNode:
        nodes = []

        def inorder(node, nodes):
            if node:
                inorder(node.left, nodes)
                nodes.append(node.val)
                inorder(node.right, nodes)

        inorder(root, nodes)

        ans = cur = TreeNode(nodes[0])
        for n in nodes[1:]:
            cur.right = TreeNode(n)
            cur = cur.right

        return ans
